jscomp/others/node_buffer.ml

```diff
@@ -26,17 +26,208 @@
 
 type t = Node.buffer 
 
-external isBuffer : 'a -> bool = "Buffer.isBuffer" 
-[@@bs.val]
+external isBuffer : 'a -> bool = "isBuffer" [@@bs.val] [@@bs.scope "Buffer"]
 
-external fromString : string -> t = "Buffer.from"
-[@@bs.val]
+external fromString : string -> t = "from" [@@bs.val] [@@bs.scope "Buffer"]
 
-external fromStringWithEncoding : string -> ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary  | `hex ] [@bs.string]) -> t = "from"
+external fromArray : int array -> t = "from" [@@bs.val] [@@bs.scope "Buffer"]
+
+external fromBuffer : t -> t = "from" [@@bs.val] [@@bs.scope "Buffer"]
+
+external fromArrayBuffer : Js_typed_array2.array_buffer -> ?offset:int -> ?length:int -> unit -> t = "from" [@@bs.val] [@@bs.scope "Buffer"]
+
+external fromStringWithEncoding :
+  string ->
+  ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string])
+  -> t = "from"
+ [@@bs.val] [@@bs.scope "Buffer"]
+
+external alloc : 
+  int ->
+  ?fill:([ `String of string | `Integer of int | `Buffer of t ] [@bs.unwrap]) ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) -> 
+  unit -> t = "alloc"
+[@@bs.val] [@@bs.scope "Buffer"]
+
+external allocUnsafe : int -> t = "allocUnsafe"
+[@@bs.val] [@@bs.scope "Buffer"]
+
+external allocUnsafeSlow : int -> t = "allocUnsafeSlow"
+[@@bs.val] [@@bs.scope "Buffer"]
+
+external byteLength : t -> int = "byteLength"
+[@@bs.get]
+
+external byteOffset : t -> int = "byteOffset"
+[@@bs.get]
+
+external byteLengthWithEncoding : 
+  ([ `String of string | `Buffer of t ] [@bs.unwrap]) ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  unit ->
+  int = "byteLength"
+[@@bs.val] [@@bs.scope "Buffer"]
+
+external compare : t -> t -> int = "compare"
 [@@bs.val] [@@bs.scope "Buffer"]
 
-external toString : t -> string = "toString"
+external toString : t -> string = "toString" [@@bs.send]
+
+external toStringWithEncoding : 
+  t ->
+  ?encoding: ([ `ascii  | `utf8  | `utf16le  | `usc2  | `base64  | `latin1 | `binary | `hex ] [@bs.string]) ->
+  ?start: int ->
+  ?end_: int ->
+  unit -> string = "toString"
+[@@bs.send]
+
+external concat : t array -> t = "concat"
+[@@bs.val] [@@bs.scope "Buffer"]
+
+external concatWithTotalLength : t array -> int -> t = "concat"
```

Comment:
These two can also be merged into a single `concat` binding with an optional `~totalLength` parameter.

Reply:
Oops, same argument applies to `concat` since it's an existing binding–shouldn't be changed.

Reply:
Kept all original module API intact with no changes.